Timeout error on aarch64-linux (RPi 4)
```
_____________ TestFileKeyringSynchronization.test_multiple_writers _____________

self = <tests.core.util.test_file_keyring_synchronization.TestFileKeyringSynchronization testMethod=test_multiple_writers>

    @using_temp_file_keyring()
    def test_multiple_writers(self):
        num_workers = 20
        keyring_path = str(KeyringWrapper.get_shared_instance().keyring.keyring_path)
        passphrase_list = list(
            map(lambda x: ("test-service", f"test-user-{x}", f"passphrase {x}", keyring_path), range(num_workers))
        )

        # When: spinning off children to each set a passphrase concurrently
        with Pool(processes=num_workers) as pool:
            res = pool.starmap_async(dummy_set_passphrase, passphrase_list)
>           res.get(timeout=10)  # 10 second timeout to prevent a bad test from spoiling the fun

tests/core/util/test_file_keyring_synchronization.py:87:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <multiprocessing.pool.MapResult object at 0x7ff00c6e50>, timeout = 10

    def get(self, timeout=None):
        self.wait(timeout)
        if not self.ready():
>           raise TimeoutError
E           multiprocessing.context.TimeoutError

/nix/store/88xnrm94m4zph83080xp01l8lv2vqdwq-python3-3.9.6/lib/python3.9/multiprocessing/pool.py:767: TimeoutError
----------------------------- Captured stdout call -----------------------------
Failed to acquire keyring writer lock after 5 seconds. Remaining attempts: 5
Failed to acquire keyring writer lock after 5 seconds. Remaining attempts: 5
Failed to acquire keyring writer lock after 5 seconds. Remaining attempts: 5
Failed to acquire keyring writer lock after 5 seconds. Remaining attempts: 5
Failed to acquire keyring writer lock after 5 seconds. Remaining attempts: 5
Failed to acquire keyring writer lock after 5 seconds. Remaining attempts: 5
Failed to acquire keyring writer lock after 5 seconds. Remaining attempts: 5
```

--- a/pkgs/applications/blockchains/chia/default.nix
+++ b/pkgs/applications/blockchains/chia/default.nix
@@ -5,7 +5,7 @@
 , python3Packages
 }:
 
-python3Packages.buildPythonApplication rec {
+let chia = python3Packages.buildPythonApplication rec {
   pname = "chia";
   version = "1.2.5";
 
@@ -64,6 +64,10 @@ python3Packages.buildPythonApplication rec {
     pytestCheckHook
   ];
 
+  # Testsuite is expensive and non-deterministic, so it is available in
+  # passthru.tests instead.
+  doCheck = false;
+
   disabledTests = [
     "test_spend_through_n"
     "test_spend_zero_coin"
@@ -86,6 +90,10 @@ python3Packages.buildPythonApplication rec {
     export HOME=`mktemp -d`
   '';
 
+  passthru.tests = {
+    chiaWithTests = chia.overrideAttrs (_: { doCheck = true; });
+  };
+
   meta = with lib; {
     homepage = "https://www.chia.net/";
     description = "Chia is a modern cryptocurrency built from scratch, designed to be efficient, decentralized, and secure.";
@@ -93,4 +101,5 @@ python3Packages.buildPythonApplication rec {
     maintainers = teams.chia.members;
     platforms = platforms.all;
   };
-}
+};
+in chia

--- a/pkgs/applications/blockchains/chia/default.nix
+++ b/pkgs/applications/blockchains/chia/default.nix
@@ -7,14 +7,14 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "chia";
-  version = "1.2.3";
+  version = "1.2.5";
 
   src = fetchFromGitHub {
     owner = "Chia-Network";
     repo = "chia-blockchain";
     rev = version;
     fetchSubmodules = true;
-    sha256 = "sha256-nK/Zk2zgIdrRtw3+VkUXQWfI9j29XFDOR95Dvbn07eA=";
+    sha256 = "sha256-3TRjOAlZfcvDKHecagA+hXRg7HqiD6NNegi956LEXS8=";
   };
 
   patches = [
@@ -45,15 +45,18 @@ python3Packages.buildPythonApplication rec {
     clvm
     clvm-rs
     clvm-tools
+    colorama
     colorlog
     concurrent-log-handler
     cryptography
     dnspython
+    fasteners
     keyrings-cryptfile
     pyyaml
     setproctitle
     setuptools # needs pkg_resources at runtime
     sortedcontainers
+    watchdog
     websockets
   ];
 
@@ -64,6 +67,8 @@ python3Packages.buildPythonApplication rec {
   disabledTests = [
     "test_spend_through_n"
     "test_spend_zero_coin"
+    "test_default_cached_master_passphrase"
+    "test_using_legacy_keyring"
   ];
 
   postPatch = ''
@@ -72,9 +77,7 @@ python3Packages.buildPythonApplication rec {
       --replace "aiohttp==3.7.4" "aiohttp>=3.7.4" \
       --replace "sortedcontainers==2.3.0" "sortedcontainers>=2.3.0" \
       --replace "click==7.1.2" "click>=7.1.2" \
-      --replace "clvm_rs==0.1.8" "clvm_rs>=0.1.8" \
       --replace "clvm==0.9.7" "clvm>=0.9.7" \
-      --replace "bitstring==3.1.7" "bitstring>=3.1.9" \
 
     ln -sf ${cacert}/etc/ssl/certs/ca-bundle.crt mozilla-ca/cacert.pem
   '';

